environs/jujutest/livetests.go

```diff
@@ -318,72 +318,93 @@ func (t *LiveTests) TestPorts(c *gc.C) {
 	defer t.Env.StopInstances(inst2.Id())
 
 	// Open some ports and check they're there.
-	err = inst1.OpenPorts("1", []network.PortRange{{67, 67, "udp"}, {45, 45, "tcp"}, {80, 100, "tcp"}})
+	err = inst1.OpenPorts(
+		"1", network.RulesFromPortRanges([]network.PortRange{{67, 67, "udp"}, {45, 45, "tcp"}, {80, 100, "tcp"}}...))
```

Comment:
Yeah, I had similar thoughts and couldn't convince myself that constructing IngressRule struct directly was any better. I expect the RulesFromPortRanges method to go away once all providers are converted across. Let's leave for now and it can be revisited in the followup work.